### PR TITLE
Fixes, improvements and update to rox-react-native

### PR DESCRIPTION
**Pull request by g-guirado**
While awesome, the initial type definition was made based on reverse engineering and improper documentation. This inevitably led to a few mistakes and an incomplete definition.
Documentation is now available, new package version has been published.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://support.rollout.io/docs/reactnative
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/rox-react-native/index.d.ts [hunk 1]
```diff
@@ -40,25 +36,26 @@ export function setup(
...
-        experiment: RoxExperiment,
-        context: unknown
-    ): void;
+    // https://support.rollout.io/docs/reactnative#section-using-the-impressionhandler-option
+    impressionHandler?(reporting: RoxReporting, experiment: RoxExperiment, context: unknown): void;
     platform?: string;
-    freeze?: 'untilLaunch' | 'none';
+    freeze?: FreezeOptions;
+    disableNetworkFetch?: boolean;
+    devModeSecret?: string;
```
Comment (ahanriat) [inline types/rox-react-native/index.d.ts]: Can you add the `AsyncStorage` option as well 🙏 
https://support.rollout.io/docs/reactnative#section--asyncstorage-

Comment (g-guirado) [reply types/rox-react-native/index.d.ts]: Done, with type 'any' though.

Comment (ahanriat) [reply types/rox-react-native/index.d.ts]: Thanks 🙏

**Comment by g-guirado** [comment]
@ahanriat Please don't forget to explicitly approve this PR once you're OK with it. It will make it move faster toward merge. Thanks!

**Review by ahanriat — APPROVED** [review]

**Comment by ahanriat** [comment]
@g-guirado Done ✅ 
Thanks for submitting this PR 🙏

**Pull request merged**